File: leetcode/剑指/剑指.10.斐波那契数列.go
var Fib map[int]int = map[int]int{0: 0, 1: 1}
func fib(n int) int {
    if _, ok := Fib[n];ok {
        return Fib[n];
    }
    Fib[n] = (fib(n - 1) + fib(n - 2)) % 1000000007;
    return Fib[n];
}

File: leetcode/剑指/剑指.10.斐波那契数列.py
class Solution:
    FibNums = {0: 0, 1: 1}
    def fib(self, n: int) -> int:
        if n in self.FibNums:
            return self.FibNums[n]
        self.FibNums[n] = (self.fib(n - 1) + self.fib(n - 2)) % 1000000007
        return self.FibNums[n]